test/irb/test_cmd.rb

```diff
@@ -469,6 +469,116 @@ def test_show_source_method
       assert_match(%r[/irb\/init\.rb], out)
     end
 
+    def test_show_source_method_s
+      code = <<~RUBY
+        class Baz
+          def foo
+          end
+        end
+      
+        class Bar < Baz
+          def foo
+            super
+          end
+        end
+      RUBY
+      File.write("#{@tmpdir}/bazbar.rb", code)
+      out, err = execute_lines(
+        "irb_load '#{@tmpdir}/bazbar.rb'\n",
+        "show_source Bar#foo -s",
+      )
+      assert_match(%r[bazbar.rb:2\n\n  def foo\n  end\n\n=> nil\n], out)
+    end
+
+    def test_show_source_method_multiple_s
+      code = <<~RUBY
+        class Baz
+          def foo
+          end
+        end
+      
+        class Bar < Baz
+          def foo
+            super
+          end
+        end
+
+        class Bob < Bar
+          def foo
+            super
+          end
+        end
+      RUBY
+      File.write("#{@tmpdir}/bazbarbob.rb", code)
+      out, err = execute_lines(
+        "irb_load '#{@tmpdir}/bazbarbob.rb'\n",
+        "show_source Bob#foo -ss",
+      )
+      assert_match(%r[bazbarbob.rb:2\n\n  def foo\n  end\n\n=> nil\n], out)
+    end
+
+    def test_show_source_method_no_instance_method
+      code = <<~RUBY
+        class Baz
+        end
+      
+        class Bar < Baz
+          def foo
+            super
+          end
+        end
+      RUBY
+      File.write("#{@tmpdir}/bazbar.rb", code)
+      out, err = execute_lines(
+        "irb_load '#{@tmpdir}/bazbar.rb'\n",
+        "show_source Bar#foo -s",
+      )
+      assert_match(%r[Error: Couldn't locate a definition for Bar#foo -s\n=> nil\n], out)
```

Comment:
Is it possible to change the message to `Couldn't locate the super definition for Bar#foo`?

Reply:
Just added this.  I moved the parsing to `ShowSource#execute" to accamodate this.